### PR TITLE
add MSYS2 build to CI

### PR DESCRIPTION
**Pull request by rfomin**

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -58,3 +58,49 @@ jobs:
...
+        with:
+          update: true
+          install: >-
+            git
+            base-devel
+            cmake
+            SDL2
+            SDL2_mixer
+            SDL2_net
+            fluidsyhth
```
Comment (fabiangreffrath) [inline .github/workflows/main.yml:87]: How about modplug?

Comment (rfomin) [reply .github/workflows/main.yml:87]: Actually fluidsyhth should depend on SDL_mixer, so it's not needed.

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -58,3 +58,50 @@ jobs:
...
+            ${{ matrix.sys }}-${{ matrix.env }}-ninja
+            ${{ matrix.sys }}-${{ matrix.env }}-cmake
+            ${{ matrix.sys }}-${{ matrix.env }}-SDL2
+            ${{ matrix.sys }}-${{ matrix.env }}-SDL2_mixer
+            ${{ matrix.sys }}-${{ matrix.env }}-SDL2_net
+
+      - uses: actions/checkout@v2
+
+      - name: Configure
+        run: cmake -B build -G "Ninja"
```
Comment (fabiangreffrath) [inline .github/workflows/main.yml:93]: Can we do release builds with MSYS?

Comment (rfomin) [reply .github/workflows/main.yml:94]: Yes, I'm trying to figure it out. Not as beautiful as in the example of course 😄

**Comment by rfomin** [comment]
Now I understand: "msys pls" (https://github.com/fabiangreffrath/woof/pull/622/commits/0f9f1cade7e3a6b1df7e107c1bfd539fd9f03ec5) is the magic command 😄

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -58,3 +58,52 @@ jobs:
...
+
+    steps:
+      - name: Setup MSYS2
+        uses: msys2/setup-msys2@v2
+        with:
+          msystem: ${{ matrix.sys }}
+          update: true
+          install: >-
+            git
+            cmake
```
Comment (fabiangreffrath) [inline .github/workflows/main.yml:84]: Do we need both msys and mingw cmake?

Comment (rfomin) [reply .github/workflows/main.yml]: It seems we don't.

**Comment by fabiangreffrath** [comment]
Now that we have this, I don't think we need the cross-building rule anymore. What should be left is Linux 64 gcc and clang, MSYS gcc 32 and 64, cppcheck and probably MSVC.

**Comment by rfomin** [comment]
> What should be left is Linux 64 gcc and clang, MSYS gcc 32 and 64, cppcheck and probably MSVC.

I agree, but first I need to deal with the release. It seems to work, but my fork is very broken: https://github.com/rfomin/woof/releases/tag/woof-10.0.1

**Comment by fabiangreffrath** [comment]
Actually, we don't need binary releases for Linux, just Windows 32 and 64 bit. Linux users are supposed to either compile the source on their own or refer to distro packages.

**Comment by rfomin** [comment]
> Actually, we don't need binary releases for Linux, just Windows 32 and 64 bit. Linux users are supposed to either compile the source on their own or refer to distro packages.

Yes, we don't create any packages for Linux targets.

Eh probably I need to create new fork. I can't ran actions currently because of broken master.

**Comment by rfomin** [comment]
It works: https://github.com/rfomin/woof/releases
I'll finish tomorrow. I also think that we should have a CHANGELOG.md in the repository that we will attach to the release, maybe even edit it along the way.

**Comment by rfomin** [comment]
Let's merge this with master first, after that I'll fix MSVC. By the way, `cppcheck` never warned me about anything, but what about `clang-tidy`?

**Review by fabiangreffrath — APPROVED** [review]
Please merge this if you need to work on the master branch.

**Comment by fabiangreffrath** [comment]
> Please merge this if you need to work on the master branch.

*Squash merge

**Pull request merged**